lua/nvim-treesitter/configs.lua

```diff
@@ -20,7 +20,7 @@ local M = {}
 ---@type TSConfig
 local config = {
   modules = {},
-  sync_install = false,
+  sync_install = #vim.api.nvim_list_uis() == 0,
   ensure_installed = {},
```

Comment:
Wouldn't this cause later problems when the UI is attached later and Neovim wasn't started with `--embed`? Would it be better to perform this check when starting async install? Do we really want to forbid async install when no UIs are present?

What benefit would this have over when we're already offering a sync and a async variant (except for user not aware of TSInstallSync

Reply:
I'm not great with nvim internals, I think what we want to do is check if the ui is headless during async install, and if so, force sync so installation doesn't exit before it's finished like in the issue mentioned. 

It helps when someone tries `:TSInstall` not knowing there's a sync version